### PR TITLE
docs(react): add guide for React Compiler

### PR DESCRIPTION
**Pull request by chenjiahan**
## Summary

Add guide for using React Compiler in Rspack:

- https://react.dev/learn/react-compiler
- https://rsbuild.dev/guide/framework/react#react-compiler

## Checklist

- [x] Tests updated (or not required).
- [x] Documentation updated (or not required).

### CODE REVIEW

File: website/docs/zh/guide/tech/react.mdx [hunk 1]
```diff
@@ -133,6 +133,87 @@ module.exports = {
 - 配合 `builtin:swc-loader` 使用方式可参考：[examples/react-refresh](https://github.com/rspack-contrib/rspack-examples/tree/main/rspack/react-refresh/rspack.config.js)，使用 `swc-loader` 只需将 `builtin:swc-loader` 换为 `swc-loader` 即可。
 - 配合 `babel-loader` 的使用方式可参考：[examples/react-refresh-babel-loader](https://github.com/rspack-contrib/rspack-examples/tree/main/rspack/react-refresh-babel-loader/rspack.config.js)
 
+## React Compiler
+
+React Compiler 是 React 19 引入的一个实验性编译器，它可以自动优化你的 React 代码。
+
+在开始使用 React Compiler 之前，建议阅读 [React Compiler 文档](https://react.dev/learn/react-compiler)，以了解 React Compiler 的功能、当前状态和使用方法。
+
+:::tip
```
Comment (SyMind) [inline website/docs/zh/guide/tech/react.mdx:142]: 更改为 :::tip 提示，否则网站展示时为 TIP 这个英文

File: website/docs/en/guide/tech/react.mdx [hunk 2]
```diff
@@ -133,6 +133,87 @@ Compared to the previous approach, this method decouples the React Fast Refresh
...
+
+1. Upgrade versions of `react` and `react-dom` to 19.
+2. React Compiler currently only provides a Babel plugin, you need to install:
+
+- [@babel/core](https://www.npmjs.com/package/@babel/core)
+- [babel-loader](https://www.npmjs.com/package/babel-loader)
+- [babel-plugin-react-compiler](https://www.npmjs.com/package/babel-plugin-react-compiler)
+- [@babel/plugin-syntax-jsx](https://www.npmjs.com/package/@babel/plugin-syntax-jsx)
+
+3. Register babel-loader in your Rspack config file:
```
Comment (SyMind) [inline website/docs/en/guide/tech/react.mdx:158]: babel-loader 改为更好 `babel-loader`

**Review by SyMind — APPROVED** [review]

File: website/docs/en/guide/tech/react.mdx (hunk 2 above)
Comment (chenjiahan) [reply website/docs/en/guide/tech/react.mdx:158]: done~

File: website/docs/zh/guide/tech/react.mdx (hunk 1 above)
Comment (chenjiahan) [reply website/docs/zh/guide/tech/react.mdx:142]: I think it's okay to display English `TIP` in the Chinese document, it's a simple word and users should know it 😄

**Review by SyMind — APPROVED** [review]

**Pull request merged**